Susi/Controllers/SkillDetailViewController/SkillDetailVCMethods.swift

```diff
@@ -186,6 +175,66 @@ extension SkillDetailViewController {
             }
         }
     }
+    
+    // Share Skill Action
+    @objc func shareSkillAction() {
+        let activityViewController : UIActivityViewController = UIActivityViewController(activityItems: [ControllerConstants.ShareSkill.message, URL(string: getSkillURL(Client.APIURLs.SkillURL,(skill?.group)!,(skill?.skillName)!,(skill?.language)!)) as Any], applicationActivities: nil)
+        // For overcoming the crash in iPad
+        activityViewController.popoverPresentationController?.sourceView = self.view
+        activityViewController.excludedActivityTypes = [
+            UIActivity.ActivityType.airDrop,
+            UIActivity.ActivityType.saveToCameraRoll,
+            UIActivity.ActivityType.openInIBooks,
+            UIActivity.ActivityType.markupAsPDF]
+        self.present(activityViewController, animated: true, completion: nil)
+        
+    }
+    
+    func skillOptionBarButton() {
+        navigationItem.rightViews = [skillOptionButton]
+    }
+    
+    func ftConfig() {
+        let config = FTConfiguration.shared
+        config.backgoundTintColor = .white
+        config.borderColor = .lightGray
+        config.menuWidth = 150
+        config.menuSeparatorColor = .lightGray
+        config.menuRowHeight = 50
+        config.cornerRadius = 10
+        
+    }
+    
+    @objc func barButtonAction(_ sender: UIBarButtonItem, event: UIEvent) {
+        let user = UserDefaults.standard.dictionary(forKey: ControllerConstants.UserDefaultsKeys.user)
+        if user != nil {
+            ftConfig()
+            let cellConfig = FTCellConfiguration()
+            cellConfig.textColor = .black
+            cellConfig.textAlignment = .center
+            cellConfig.textFont = .systemFont(ofSize: 20)
+            let cellConfis = Array(repeating: cellConfig, count: 2)
+            FTPopOverMenu.showForEvent(event: event, with: menuOptionsAfterLogin, menuImageArray: nil, cellConfigurationArray: cellConfis, done: { (selectedIndex) in
+                if selectedIndex == 0 {
+                    self.shareSkillAction()
+                } else {
+                    self.reportSkillAction()
+                }
+            })
+        } else {
+            ftConfig()
+            let cellConfig = FTCellConfiguration()
+            cellConfig.textColor = .black
+            cellConfig.textAlignment = .center
+            cellConfig.textFont = .systemFont(ofSize: 20)
```

Comment:
why duplication of the code when it common for two loops. Use the common code outside the loop and conditional code in the loop.

Reply:
Done , Please review